app/services/setting_registry.rb

```diff
@@ -52,9 +65,25 @@ def []=(name, value)
     definition = find(name)
     raise ::Foreman::Exception.new(N_("Setting definition for '%s' not found, can not set"), name) unless definition
     db_record = _find_or_new_db_record(name)
-    db_record.value = value
-    db_record.save!
-    definition.value = db_record.value
+    db_record.update!(value: value)
+  end
+
+  def set_user_value(name, value)
+    definition = find(name)
+    raise ActiveRecord::RecordNotFound.new(_("Setting definition for '%s' not found, can not set") % name, Setting, name) unless definition
+    db_record = _find_or_new_db_record(name)
+    db_record.send(:set_setting_type_from_value)
+    # rubocop:disable Style/DoubleNegation
+    case (!!value == value ? 'boolean' : value.class.to_s.downcase)
+    # rubocop:enable Style/DoubleNegation
```

Comment:
It still surprises me that Ruby doesn't have a `Boolean.new(value)` method (or similar) to force it as a boolean.

That said: why isn't checking for trueclass / falseclass no longer the right thing?

Reply:
I was looking for something shorter, this works the best from what I found and is much easier to read IMHO.
But both are fine, this is shorter. If you think checking for two classes was better, I'm ok with that as well :)

Reply:
To me that was easier to understand as a reader. I'm not used to reading `!!` and the fact Rubocop disables it by default probably also means something.

Reply:
Back to using the `trueclass` and `falseclass` now :)